protocol/x/perpetuals/keeper/perpetual_test.go

```diff
@@ -320,6 +320,69 @@ func TestModifyPerpetual_Failure(t *testing.T) {
 	}
 }
 
+func TestSetPerpetualMarketType_Success(t *testing.T) {
```

Comment:
nit: Can these tests be parameterized in a set of test cases rather than be separate tests?